### PR TITLE
Allow empty YAML files

### PR DESCRIPTION
**Pull request by jamiemccarthy**
In my testing, `parse` with a YAML file that consisted solely of a comment

```
# comment
```

would call `parse_recurse` with `psych_parse_data == nil`. This would raise a `NilError` on the expression `psych_parse_data.children`. That was a bug not caught in the test suite, which included an empty file but no comment-only file.

Initially I added a `return if psych_parse_data.nil?` to fix the bug. I had intended to stop there, but on investigation, I convinced myself that empty YAML files are valid and should not break lint:

> "A YAML stream consists of zero or more documents" [9.1](https://yaml.org/spec/1.2.2/#92-streams)

So I rewrote the tests to not consider empty files invalid.

Because of that, I'd like to submit this PR as an alternative to https://github.com/shortdudey123/yamllint/pull/41 . (Tagging @LoXeras for a second opinion.)

I didn't address the rubocop errors, but I see #39 handles that. Apart from that, the test suite for this branch passes for me locally.

I recognize it's been a while since this gem was updated. I only spent a little while googling for alternatives, but from what I saw, this gem seemed like the best native-ruby YAML linter. I like your configuration options and test suite. So I hope you'll consider this for an update.

### CODE REVIEW

File: spec/linter_spec.rb [hunk 1]
```diff
@@ -63,11 +58,7 @@
...
   end
 
-  it 'should be unhappy with an empty YAML file' do
-    expect(linter.check(spec_data('empty.yaml'))).to be(false)
-  end
-
-  it 'should be unhapy with a YAML file full of spaces' do
-    expect(linter.check(spec_data('spaces.yaml'))).to be(false)
+  it 'should be happy with a YAML file full of spaces' do
+    expect(linter.check(spec_data('spaces.yaml'))).to be(true)
```
Comment (shortdudey123) [inline spec/linter_spec.rb]: Can you rename the file and add valid_ to the start of the file name?  Then remove the references to spaces.yaml and empty.yaml from the rakefile.

Comment (shortdudey123) [inline spec/linter_spec.rb]: Can you rename the file and add valid_ to the start of the file name?  Then remove the references to spaces.yaml and empty.yaml from the rakefile.

Comment (jamiemccarthy) [reply spec/linter_spec.rb]: Sure thing. How does that look?

**Comment by shortdudey123** [comment]
This looks good now.  I will merge this and close #41.  Thank you for the PR!

**Pull request merged**